### ISSUE TITLE
Value Error: Error when checking target: expected dense_28 to have shape (7,) but got array with shape (1,)

### ISSUE THREAD

**Issue by HussamCheema** [keras-team/keras#11766]
I have 7 label categories in x_test and y_test. So, Nodes of last Dense Layer should be 7 instead of 1. Why its given me this error? 
x_train and x_test have 1893 images and corresponding to those images I have 471 y_train and y_test labels ranging from 1 to 7.

```
model = Sequential()
model.add(Conv2D(kernel_size=(3,3),filters=16,input_shape=(256, 256, 3),activation="relu",padding="valid"))
model.add(Conv2D(kernel_size=(3,3),filters=16,activation="relu",padding="same"))
model.add(MaxPooling2D(pool_size=(3,3),strides=(1,1)))
model.add(Conv2D(kernel_size=(3,3),filters=8,activation="relu",padding="same"))
model.add(Conv2D(kernel_size=(2,2),filters=8,activation="relu",padding="same"))
model.add(Flatten())
model.add(Dense(100,activation="relu"))
model.add(Dense(7,activation='sigmoid'))
model.summary()

model.compile(loss=keras.losses.categorical_crossentropy,
              optimizer=keras.optimizers.Adam(lr=0.001),
              metrics=['accuracy'])

X = model.fit(x_train, y_train,
          batch_size=batch_size,
          epochs=epochs,
          verbose=1,
          validation_data=(x_test, y_test))

plt.plot(X.history['loss'])
plt.plot(X.history['val_loss'])
plt.title('model loss')
plt.ylabel('loss')
plt.xlabel('epoch')
plt.legend(['train', 'validation'], loc='upper left')
plt.show()

score = model.evaluate(x_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])
```
